mlflow/sklearn/__init__.py

```diff
@@ -994,3 +986,102 @@ def patched_fit(original, self, *args, **kwargs):
                 safe_patch(
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
+
+
+def log_eval_metrics(*, model=None, model_uri=None, X, y_true, prefix, sample_weight=None):
+    """
+    Computes and logs evaluation metrics (and artifacts) for the given model and labeled dataset.
+    The metrics/artifacts mirror what is auto-logged when training a model
+    (see mlflow.sklearn.autolog).
+
+    ** Example **
+
+    .. code-block:: python
+
+        from sklearn.linear_model import LinearRegression
+        import mlflow
+
+        # enable autologging
+        mlflow.sklearn.autolog()
+
+        # prepare training data
+        X = np.array([[1, 1], [1, 2], [2, 2], [2, 3]])
+        y = np.dot(X, np.array([1, 2])) + 3
+
+        # prepare evaluation data
+        X_eval = np.array([[3, 3], [3, 4]])
+        y_eval = np.dot(X_eval, np.array([1,2])) + 3
+
+        # train a model
+        model = LinearRegression()
+        with mlflow.start_run() as run:
+            model.fit(X, y)
+            mlflow.log_eval_metrics(model=model, X=X_eval, y_true=y_eval, prefix="val_")
+
+
+    Each metric's and artifact's name is prefixed with `prefix`, e.g., in the previous example the
+    metrics and artifacts are named 'eval_XXXXX'. Note that training-time metrics are auto-logged
+    as 'training_XXXXX'.
+
+    The run under which to log the metrics/artifacts is chosen as follows:
+    - Under the active run if one exists
+    - Or, under the same run as the model_uri if the latter is of the form 'runs:/...'
+    - Or, under a new run
+    The chosen run is in the return value of the method.
+
+    :param model: The model to be evaluated. Exactly one model_uri or model must be specified.
+    :param model_uri: The URI of the model to be evaluated. Exactly one model_uri or model must be
+                    specified.
+    :param X: The features for the evaluation dataset.
+    :param y_true: The labels for the evaluation dataset.
+    :param prefix: Prefix used to name metrics and artifacts.
+    :param sample_weight: Per-sample weights to apply in the computation of metrics/artifacts.
+    :return: A tuple (run_id, metrics, artifacts) where run_id is the run under which
```

Comment:
If the run is created as part of this method (see the case for creating a new run) then it seems useful to return it so that the caller can access everything that got logged. If we remove run_id from the return then we need to return artifacts + potentially more stuff that we end up logging in the future. Maybe safer to return run_id for now + metrics and omit everything else (e.g., artifacts)?

Reply:
See my above comment about leaving the newly created run open.

Reply:
Done.